### PR TITLE
Fix/fetching accommodations

### PR DESCRIPTION
**Pull request by melievkh**

### CODE REVIEW

File: src/components/Card/Card.tsx [hunk 1]
```diff
@@ -1,69 +1,72 @@
...
+  const handleToggleWishlist = () => {
+    if (!isLoggedIn) {
+      showAlert('warning', { message: 'You should login first!' });
+      return;
+    }
+
     setPressed(!pressed);
+    if (pressed) {
+      onRemoveFromWishlist(item.id);
+    } else {
```
Comment (gitkoshelew) [inline src/components/Card/Card.tsx]: following style usually more convenient to read  
 if (pressed) {
      onRemoveFromWishlist(item.id);
      return
    } 
   onAddedToWishlist(item.id);

File: src/components/Card/Card.tsx [hunk 2]
```diff
@@ -1,69 +1,72 @@
...
   const [pressed, setPressed] = useState<boolean>(false);
 
-  const handlePressed = () => {
+  const handleToggleWishlist = () => {
+    if (!isLoggedIn) {
+      showAlert('warning', { message: 'You should login first!' });
+      return;
+    }
+
     setPressed(!pressed);
```
Comment (gitkoshelew) [inline src/components/Card/Card.tsx]: setPressed(!pressed) should be invoked on function bottom, cuz 'set' may work asyncroniously

File: src/components/Modals/ExploreModals/FilterModal/FilterModal.tsx [hunk 3]
```diff
@@ -75,38 +67,38 @@ const FilterModal = () => {
           <View style={styles.filterContentContent}>
             <Input
               style={styles.input}
-              value={formValues.minPrice?.toString()}
+              // value={formValues.minPrice?.toString()}
```
Comment (gitkoshelew) [inline src/components/Modals/ExploreModals/FilterModal/FilterModal.tsx]: remove it at all if you're not gonna use it

File: src/utils/location/index.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+const getPlaceDetails = async (placeId: string) => {
+  const apiUrl = `https://maps.googleapis.com/maps/api/place/details/json?place_id=${placeId}&key=${GOOGLE_API_KEY}`;
+
+  try {
+    const response = await fetch(apiUrl);
+    const data = await response.json();
+
+    if (data.status === 'OK') {
+      return data.result;
+    } else {
```
Comment (gitkoshelew) [inline src/utils/location/index.ts]: you don't need else when you return smth inside 'if'

File: src/axios/axiosInstance.ts [hunk 5]
```diff
@@ -50,6 +43,7 @@ axiosInstance.interceptors.response.use(
 
     if (config.url === ENDPOINTS.refresh(userId) && data?.error?.statusCode === 401) {
       await SecureStore.deleteItemAsync(SecureStorageKey.REFRESH_TOKEN);
+      // store.dispatch(accountActions.reset());
```
Comment (adadychanka) [inline src/axios/axiosInstance.ts]: Is it an unnecessary comment?

File: src/components/Card/Card.tsx [hunk 6]
```diff
@@ -1,69 +1,72 @@
...
-  address,
-}: CardProps) => {
+const Card = ({ item, onAddedToWishlist, onRemoveFromWishlist }: Props) => {
   const navigation = useNavigation<NavigationProp<RootStackParamList>>();
+  const isLoggedIn = useSelector(getIsLoggedIn);
   const [pressed, setPressed] = useState<boolean>(false);
 
-  const handlePressed = () => {
+  const handleToggleWishlist = () => {
+    if (!isLoggedIn) {
```
Comment (adadychanka) [inline src/components/Card/Card.tsx:27]: I think it's better to extract this part of the logic related to isLoggedIn because having it here makes this component smarter than it should be

Comment (adadychanka) [reply src/components/Card/Card.tsx:27]: https://en.wikipedia.org/wiki/Single_responsibility_principle

Comment (adadychanka) [reply src/components/Card/Card.tsx:27]: you always can have a callback that is passed to this component in which you put this warning

File: src/components/ImageCarousel/ImageCarousel.tsx [hunk 7]
```diff
@@ -1,12 +1,13 @@
-import React from 'react';
-import { View, Image, StyleSheet } from 'react-native';
+import { View, Image } from 'react-native';
 import Swiper from 'react-native-swiper';
 
-interface ImageCarouselProps {
+import { styles } from './ImageCarousel.style';
+
+interface Props {
```
Comment (adadychanka) [inline src/components/ImageCarousel/ImageCarousel.tsx]: Sometimes it might be useful to have properly named props type.

File: src/components/Modals/centerModals/DatePicker/DatePicker.tsx [hunk 8]
```diff
@@ -2,61 +2,56 @@ import { useState } from 'react';
...
 
-const DateTimePicker = ({ error, initialValue, label, minDate, maxDate, onDateChange }: Props) => {
+const DateTimePicker = ({ initialValue, label, minDate, maxDate, onDateChange }: Props) => {
   const colors = useSelector(getColors);
-  const initialDate = getInitialDate({ initialValue });
+  const initialDate = getInitialDate(initialValue);
 
   const [showDatePicker, setShowDatePicker] = useState<boolean>(false);
-  const [selectedDate, setSelectedDate] = useState<string>(initialDate);
+  const [selectedDate, setSelectedDate] = useState<string | undefined>(initialDate);
```
Comment (adadychanka) [inline src/components/Modals/centerModals/DatePicker/DatePicker.tsx:28]: Why do you need a union with undefined type here?

File: src/components/inputs/PlacesInput/PlacesInput.styles.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
+import { StyleSheet } from 'react-native';
+
+export const styles = StyleSheet.create({
+  container: {},
```
Comment (adadychanka) [inline src/components/inputs/PlacesInput/PlacesInput.styles.ts]: if you don't need styles for container, remove it

File: src/screens/bottomTabs/Wishlist/Wishlist.tsx [hunk 10]
```diff
@@ -50,7 +56,8 @@ const Wishlist = ({ navigation }: Props) => {
           renderItem={({ item }) => (
             <WishlistItem
               wishlistDetails={item}
-              onDelete={() => handleDelete(item.accommodation.id)}
+              onDelete={() => handleRemoveFromWishlist(item.accommodation.id)}
```
Comment (adadychanka) [inline src/screens/bottomTabs/Wishlist/Wishlist.tsx]: https://github.com/VentionCapstone/Mobile/pull/18/files#diff-e1ebb7ba104a124413913dd9589d9f6a5324b703c9f8c146f774234d0528f4f0L14

You can add `accomodationId` argument to `onDelete` callback prop to avoid passing inline functions here

**Review by adadychanka — CHANGES_REQUESTED** [review]
need a small polishing

File: src/axios/axiosInstance.ts (hunk 5 above)
Comment (melievkh) [reply src/axios/axiosInstance.ts]: No, it isnt)

File: src/components/Modals/centerModals/DatePicker/DatePicker.tsx (hunk 8 above)
Comment (melievkh) [reply src/components/Modals/centerModals/DatePicker/DatePicker.tsx:28]: Because initialDate value sometimes come as undefined when editing accommodation or kinda related operations

File: src/axios/axiosInstance.ts [hunk 11]
```diff
@@ -18,18 +18,11 @@ const endpointsWithoutToken = [ENDPOINTS.signin, ENDPOINTS.signup];
 axiosInstance.interceptors.request.use(
   async (config: InternalAxiosRequestConfig) => {
     const accessToken = await SecureStore.getItemAsync(SecureStorageKey.ACCESS_TOKEN);
-    const refreshToken = await SecureStore.getItemAsync(SecureStorageKey.ACCESS_TOKEN);
 
     if (config.url && accessToken && !endpointsWithoutToken.includes(config.url)) {
       config.headers.Authorization = `Bearer ${accessToken}`;
     }
 
-    if (config.url === ENDPOINTS.signout && refreshToken) {
```
Comment (gitkoshelew) [inline src/axios/axiosInstance.ts:27]: this kind of changes should be merged in another pr/fix

File: src/components/Modals/ExploreModals/FilterModal/FilterModal.styles.ts [hunk 12]
```diff
@@ -1,60 +1,45 @@
 import { StyleSheet } from 'react-native';
-import { GREY_500, WHITE } from 'src/styles';
+import { title1, title2, LEVEL_1 } from 'src/styles';
+import { HORIZONTAL_12_PERCENT, HORIZONTAL_16_PERCENT } from 'src/utils';
```
Comment (gitkoshelew) [inline src/components/Modals/ExploreModals/FilterModal/FilterModal.styles.ts]: better not to use dimensions in variable names. change to PADDING_HORIZONTAL_MAIN ets.. or any suitable name

File: src/axios/axiosInstance.ts (hunk 11 above)
Comment (melievkh) [reply src/axios/axiosInstance.ts:27]: yeap, got it)

**Review by gitkoshelew — APPROVED** [review]

**Review by javokhirm — APPROVED** [review]
